timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -406,7 +394,8 @@ void __device__ v_nonbonded_unified(
         // (ytz): note that d2ij must be *strictly* less than cutoff_squared. This is because we set the
         // non-interacting atoms to exactly real_cutoff*real_cutoff. This ensures that atoms who's 4th dimension
         // is set to cutoff are non-interacting.
-        if (d2ij < cutoff_squared && atom_j_idx > atom_i_idx && atom_j_idx < N && atom_i_idx < N) {
+        if (d2ij < cutoff_squared && atom_i_idx < N &&
+            (NR == 0 && atom_j_idx > atom_i_idx && atom_j_idx < N || NR != 0 && atom_j_idx < NC)) {
```

Comment:
ugh

Reply:
agreed this is bad -- attempted to clarify a bit in 05b1b35